webpack/assets/javascripts/react_app/components/common/forms/InputFactory.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormControl } from 'patternfly-react';
+
+import { noop } from '../../../common/helpers';
+import AutoComplete from '../../AutoComplete';
+import DateTimePicker from '../DateTimePicker/DateTimePicker';
+import DatePicker from '../DateTimePicker/DatePicker';
+import OrderableSelect from './OrderableSelect';
+import TimePicker from '../DateTimePicker/TimePicker';
+
+const inputComponents = {
+  autocomplete: AutoComplete,
+  date: DatePicker,
+  dateTime: DateTimePicker,
+  orderableSelect: OrderableSelect,
+  time: TimePicker,
+};
+
+export const registerInputComponent = (name, Component) => {
+  inputComponents[name] = Component;
+};
+
+const InputFactory = ({ type, ...controlProps }) => {
```

Comment:
Not a blocker but how does it align with @xprazak2 workaround migrating to `formik`?